### PR TITLE
Feature/improve open api generation

### PR DESCRIPTION
**Pull request by michelu89**

### CODE REVIEW

File: core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts [hunk 1]
```diff
@@ -31,88 +32,216 @@ export interface OpenApi {
...
+    const propertyName = fileType === 'json' ? 'jsonProperties' : 'ymlProperties';
+    this.form.patchValue({[propertyName]: content});
+  }
+
+  private getFileType(file: File): 'json' | 'yml' {
+    if (file.name.endsWith('.json')) {
+      return 'json';
+    } else if (file.name.endsWith('.yaml') || file.name.endsWith('.yml')) {
+      return 'yml';
+    }
```
Comment (larisaVasile) [inline core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts:190]: maybe we can have an enum with 'yml', 'yaml' and json

File: core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts [hunk 2]
```diff
@@ -31,88 +32,216 @@ export interface OpenApi {
...
+
+    throw new Error('Unsupported file type');
+  }
```
Comment (larisaVasile) [inline core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts:193]: this 'Unsupported file type ' can be translated

**Review by larisaVasile — APPROVED** [review]

File: core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts (hunk 2 above)
Comment (michelu89) [reply core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts:193]: At the moment we do not yet have a definition of how to deal with Console output langauges.

File: core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts (hunk 1 above)
Comment (michelu89) [reply core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts:190]: I have considered the abstraction, but given that its current application is limited only to our immediate needs, I recommend not implementing it at this stage.

**Pull request merged**